exports/taskfiles/utils/cmake.yaml

```
@@ -193,6 +211,43 @@ tasks:
       - task: "install"
         vars:
           BUILD_DIR: "{{.BUILD_DIR}}"
+          CMAKE_SETTINGS_DIR: "{{.CMAKE_SETTINGS_DIR}}"
           EXTRA_ARGS:
             ref: ".INSTALL_ARGS"
           INSTALL_PREFIX: "{{.INSTALL_PREFIX}}"
+          NAME: "{{.NAME}}"
+
+  # Setup all CMake dependencies for a project by:
+  #   1. Create a directory to contain all CMake settings files (CMAKE_SETTINGS_DIR).
+  #   2. Install all dependencies by running DEP_TASK.
+  #   3. Include each dependency's settings file in a combined settings file (CMAKE_SETTINGS_FILE)
+  #      for use inside a CMake project.
+  #
+  # @param {string} CMAKE_SETTINGS_DIR A directory path to write CMake settings files to.
+  # @param {string} DEP_TASK A task to run that will install all dependencies.
+  #   - The task name must be qualified from the root of the project.
+  #   - The task must not require any arguments (to use a task with arguments create a new task that
+  #   calls the original with any arguments set).
+  #   - Dependencies must write their settings file to CMAKE_SETTINGS_DIR to be included into the
+  #   combined settings file (CMAKE_SETTINGS_FILE).
+  # @param {string} [CMAKE_SETTINGS_FILE]
+  # @param {string} [CMAKE_SETTINGS_FILE={{.CMAKE_SETTINGS_DIR}}/settings.cmake] The file to write
+  # that includes each dependency's settings file.
+  setup-deps:
+    internal: true
+    label: "{{.TASK}}:{{.CMAKE_SETTINGS_DIR}}-{{.DEP_TASK}}"
+    vars:
+      CMAKE_SETTINGS_FILE: >-
+        {{default (printf "%s/settings.cmake" .CMAKE_SETTINGS_DIR) .CMAKE_SETTINGS_FILE}}
+    requires:
+      vars: ["CMAKE_SETTINGS_DIR", "DEP_TASK"]
+    cmds:
+      - "rm -rf {{.CMAKE_SETTINGS_DIR}}"
+      - "mkdir -p {{.CMAKE_SETTINGS_DIR}}"
+      - task: "::{{.DEP_TASK}}"
+      - >-
+        for file in {{.CMAKE_SETTINGS_DIR}}/*.cmake; do
+          if [ "$file" != "{{.CMAKE_SETTINGS_FILE}}" ]; then
```

```suggestion
          if [[ "$file" != "{{.CMAKE_SETTINGS_FILE}}" ]]; then
```

Unless you want more portability rather than fewer nuances?